src/generator/type_inference_visitor.ts

```diff
@@ -152,6 +160,18 @@ class TypeInferenceVisitor extends Visitor {
                 throw new Error(`Unknown type for right side of assignment to ${node.left.name}`);
             }
 
+            this.expressionTypes.set(node.left, rightSideType);
+            this.variableTypes.push(rightSideType);
+        } else if (isMemberExpression(node.left)) {
+            super.visit(node.left);
+
+            const rightSideType = this.expressionTypes.get(node.right);
+
+            if (rightSideType === undefined) {
+                throw new Error(`Unknown type for right side of
+                    assignment to ${(node.left.object as Identifier).name}`);
```

Comment:
`node.left.object` could also be a non-identifier and this would result in `undefined`